vortex-array/src/array/constant/mod.rs

```diff
@@ -59,23 +57,25 @@ impl ConstantArray {
         })
     }
 
-    pub fn scalar(&self) -> &Scalar {
-        &self.metadata().scalar
+    pub fn scalar_value(&self) -> &ScalarValue {
+        &self.metadata().scalar_value
+    }
+
+    /// Construct an owned [`vortex_scalar::Scalar`] with a value equal to [`Self::scalar_value()`].
+    pub fn owned_scalar(&self) -> Scalar {
```

Comment:
I don't love that we now have to construct a Scalar in several places just to do comparisons. I suspect we could, in principle, transform all of these to some sort of `self.dtype().XXX(self.scalar_value(), other.value())`; however, I didn't tackle that in this PR.